### ISSUE TITLE
How exactly does the Gram matrix get the style of the image? 

### ISSUE THREAD

**Issue by messierandromeda** [leongatys/PytorchNeuralStyleTransfer#19]
And what is the reason that the Gram matrix works so well for style transfer?

**Comment by fdan** [comment]
this is a good article: https://arxiv.org/abs/1701.01036

Informally you can think of it as a sort of hash or fingerprint of the activations of a layer, which the optimiser has to "discover".  Because there's any number of structureless jumbled style representations which would match this hash, the optimiser is statistically more likely to converge on them than the original artwork.  The content loss keeps a balance so that details of the subject etc are also preserved.

**Comment by messierandromeda** [comment]
Just making sure that I've understood this correctly, is the hash or fingerprint referring to matching the features in a layer of a CNN (feature correlations) and the structureless jumbled style representations are the texture information of the style image, but not the global arrangement? So is the neural network trying to match the two gram matrices between the input image and the style image to minimize the loss between them?

**Comment by fdan** [comment]
Exactly, and other representations work as well or better than gram matrices - histograms for example https://arxiv.org/abs/1701.08893
